mobile/redux/features/signupSlice.ts

```diff
@@ -0,0 +1,285 @@
+import { PayloadAction, createAsyncThunk, createSlice } from "@reduxjs/toolkit";
+import { GnoNativeApi, KeyInfo } from "@gnolang/gnonative";
+import { ThunkExtra } from "redux/redux-provider";
+import { Alert } from "react-native";
+
+export enum SignUpState {
+  user_exists_on_blockchain_and_local_storage = 'user_exists_on_blockchain_and_local_storage',
+  user_exists_under_differente_key = 'user_exists_under_differente_key',
+  user_exists_only_on_local_storage = 'user_exists_only_on_local_storage',
+  user_already_exists_on_blockchain_under_different_name = 'user_already_exists_on_blockchain_under_different_name',
+  user_already_exists_on_blockchain = 'user_already_exists_on_blockchain',
+  account_created = 'account_created',
+}
+
+export interface CounterState {
+  signUpState?: SignUpState
+  newAccount?: KeyInfo;
+  existingAccount?: KeyInfo;
+  loading: boolean;
+  progress: string[];
+}
+
+const initialState: CounterState = {
+  signUpState: undefined,
+  newAccount: undefined,
+  existingAccount: undefined,
+  loading: false,
+  progress: [],
+};
+
+interface SignUpParam {
+  name: string;
+  password: string;
+  phrase: string;
+}
+
+type SignUpResponse = { newAccount?: KeyInfo, existingAccount?: KeyInfo, state: SignUpState };
+
+/**
+ * This thunk checks if the user is already registered on the blockchain and/or local storage.
+ * The output is a state that will be used to decide the next step (signUpState).
+ *
+ * CASE 1.0: The user is: local storage (yes), blockchain (yes), under same name (yes) and address (yes), it will offer to do normal signin or choose new name.
+ * CASE 1.1: The user is: local storage (yes), blockchain (yes), under same name (yes) and address (no),  it will offer to delete the local storage.
+ * CASE 1.2: The user is: local storage (yes), blockchain (no),  under same name (---) and address (--),  it will offer to delete the local storage.
+ *
+ * CASE 2.0: The user is: local storage (no), blockchain (yes), under same name (no) and address (yes)
+ * CASE 2.1: The user is: local storage (no), blockchain (yes), under same name (no) and address (no)
+ *
+ * CASE 3.0: The user is: local storage (no), blockchain (no), under same name (---) and address (--), it will proceed to create the account.
+ *
+ * ref: https://github.com/gnolang/dsocial/issues/72
+ */
+export const signUp = createAsyncThunk<SignUpResponse, SignUpParam, ThunkExtra>("user/signUp", async (param, thunkAPI) => {
+
+  const { name, password, phrase } = param;
+  const gnonative = thunkAPI.extra.gnonative as GnoNativeApi;
+  let blockchainResult;
+
+  thunkAPI.dispatch(addProgress(`checking if "${name}" is already registered on the blockchain.`))
+  const result = await gnonative.qEval("gno.land/r/demo/users", `GetUserByName("${name}")`);
+  thunkAPI.dispatch(addProgress(`response: "${result}"`))
```

Comment:
Hi @iuricmp . See my Discord DM to you. If you use `GetJsonUserByName`, it returns `("" string)` if the user does not exist in r/demo/users. And if it does exist, then the JSON `result` has the address.